### PR TITLE
changes based on feedback from @stof in #5739

### PR DESCRIPTION
**Pull request by alcohol**

### CODE REVIEW

File: res/composer-repository-schema.json [hunk 1]
```diff
@@ -62,29 +62,34 @@
                     "type": "string",
                     "description": "Normalized version, optional but can save computational time on client side."
                 },
-                "source": { "type": "object" },
-                "dist": { "type": "object" },
+                "autoload": { "type": "object" },
+                "require": { "type": "object" },
```
Comment (stof) [inline res/composer-repository-schema.json]: you should add more validation on them (links are maps with strings as values), and ``autoload`` has a specific schema too

File: res/composer-repository-schema.json [hunk 2]
```diff
@@ -62,29 +62,34 @@
                     "type": "string",
                     "description": "Normalized version, optional but can save computational time on client side."
                 },
-                "source": { "type": "object" },
-                "dist": { "type": "object" },
```
Comment (stof) [inline res/composer-repository-schema.json]: shouldn't these be listed explicitly for packages ?

Comment (alcohol) [inline res/composer-repository-schema.json]: They are now in the package-base schema. Or do I misunderstand your comment?

File: res/composer-repository-schema.json (hunk 1 above)
Comment (alcohol) [inline res/composer-repository-schema.json]: Couldn't they also contain arrays as values? Not just strings?

**Pull request merged**